### PR TITLE
testcase/kernel/tc_libc_timer : Add CONFIG specific buffer check line

### PR DESCRIPTION
**Pull request by hasw7569**
If CONFIG_LIBC_LOCALTIME or CONFIG_TIME_EXTENDED is turned on.
The strftime function that receives the %a and %A parameter stores the value associated with st_time.tm_wday in buffer instead of "Day".
So I added a #if line to handle this.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [b2aeee12f4e9d5e1aad3e5aadb708eb6862e7cb4] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [b2aeee12f4e9d5e1aad3e5aadb708eb6862e7cb4] - Code Rule Check OK.

File: apps/examples/testcase/le_tc/kernel/tc_libc_timer.c [hunk 1]
```diff
@@ -321,6 +321,9 @@ static void tc_libc_timer_strftime(void)
 	st_time.tm_mday = 19;
 	st_time.tm_mon = 5;
 	st_time.tm_year = 2017 - TM_YEAR_BASE;
+#if defined(CONFIG_LIBC_LOCALTIME) || defined(CONFIG_TIME_EXTENDED)
+	st_time.tm_wday = 0;
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/kernel/tc_libc_timer.c]: How about matching this value based on the above?

Comment (hasw7569) [reply apps/examples/testcase/le_tc/kernel/tc_libc_timer.c]: I agree with your opinion.
I will change the value of st_time.tm_wday to 5 (friday)

**Comment by seinfra** [comment]
Target : [dc2de8e7a1676dc0ce7b26fa4632d8a1c142b153] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [dc2de8e7a1676dc0ce7b26fa4632d8a1c142b153] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [82e5c9460bb6a8bd8cfe71aa7fd943896a87b395] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [82e5c9460bb6a8bd8cfe71aa7fd943896a87b395] - Code Rule Check OK.

File: lib/libc/time/lib_strftime.c [hunk 2]
```diff
@@ -127,6 +127,8 @@ static const char *const g_monthname[12] = {
  *   ter,  and  terminated  by  a  conversion  specifier  character, and are
  *   replaced in s as follows:
  *
+ *   %a     A three-letter abbreviation for the day of the week.
+ *   %A     The full name for the day of the week.
```
Comment (sunghan-chang) [inline lib/libc/time/lib_strftime.c]: Please split this as different commit.
